File: app/src/main/java/com/vano/myrestaurant/controller/activity/DrinkDetailActivity.java
package com.vano.myrestaurant.controller.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

import com.vano.myrestaurant.R;
import com.vano.myrestaurant.model.entity.Drink;
import com.vano.myrestaurant.model.service.DrinkService;
import com.vano.myrestaurant.model.util.ActivityUtil;

public class DrinkDetailActivity extends AppCompatActivity {

    public static final String ID = "id";

    private final DrinkService drinkService;

    public DrinkDetailActivity() {
        drinkService = new DrinkService(this);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_drink_detail);

        ActivityUtil.configureActionBar(this, getString(R.string.drink_title));
        final Drink drink = drinkService.read(getExtraId());

        setActivityInfo(drink);
    }

    private int getExtraId() {
        final Intent intent = getIntent();

       return intent.getIntExtra(ID, 0);
    }

    @SuppressLint("SetTextI18n")
    private void setActivityInfo(Drink drink) {
        ImageView photo = findViewById(R.id.photo);
        TextView name = findViewById(R.id.name);
        TextView description = findViewById(R.id.description);
        TextView volume = findViewById(R.id.volume);

        photo.setImageResource(drink.getResourceId());
        photo.setContentDescription(drink.getName());
        name.setText(drink.getName());
        description.setText(drink.getDescription());
        volume.setText(drink.getVolume() + getString(R.string.volume_measure_unit));
    }
}

File: app/src/main/java/com/vano/myrestaurant/controller/activity/DrinkDetailActivity.kt
package com.vano.myrestaurant.controller.activity

import androidx.appcompat.app.AppCompatActivity
import com.vano.myrestaurant.model.service.DrinkService
import android.os.Bundle
import com.vano.myrestaurant.R
import com.vano.myrestaurant.model.util.ActivityUtil
import com.vano.myrestaurant.model.entity.Drink
import android.annotation.SuppressLint
import android.content.Intent
import android.view.Menu
import android.view.MenuItem
import com.vano.myrestaurant.databinding.ActivityDrinkDetail1Binding

class DrinkDetailActivity : AppCompatActivity() {

    companion object {
        const val ID = "id"
        const val DEFAULT_ID_VALUE = 0
    }

    private val extraId: Int
        get() {
            val intent = intent
            return intent.getIntExtra(ID, DEFAULT_ID_VALUE)
        }

    private var drinkService: DrinkService? = null

    private var binding: ActivityDrinkDetail1Binding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDrinkDetail1Binding.inflate(layoutInflater)
        setContentView(binding?.root)

        drinkService = DrinkService(this)

        ActivityUtil.configureActionBar(this, getString(R.string.drink_title))
        val drink = drinkService?.read(extraId)

        drink?.let { setActivityInfo(it) }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.favorite -> {
                val intent = Intent(this, FavoriteFoodActivity::class.java)
                startActivity(intent)
            }
            R.id.create_order -> {
                val intent = Intent(this, OrderActivity::class.java)
                startActivity(intent)
            }
        }
        return super.onOptionsItemSelected(item)
    }

    @SuppressLint("SetTextI18n")
    private fun setActivityInfo(drink: Drink) =
        binding?.let {
            val photo = it.photo
            val name = it.name
            val description = it.description
            val volume = it.volume

            photo.setImageResource(drink.resourceID)
            photo.contentDescription = drink.name
            name.text = drink.name
            description.text = drink.description
            volume.text = drink.volume.toString() + getString(R.string.volume_measure_unit)
        }

}